src/components/App/App.jsx

```diff
@@ -1,60 +1,60 @@
-import React from 'react';
+import React, { useCallback, useEffect, useState } from 'react';
 import styles from './App.styl';
 import Header from '../Header/Header';
+import MovieHeader from '../MovieHeader/MovieHeader';
 import MainContent from '../MainContent/MainContent';
 import Footer from '../Footer/Footer';
 import WithWrapper from '../WithWrapper/WithWrapper';
 import fetchMovies from '../../utils/fetchMovies';
 import normalizeMoviesData from '../../utils/normalizeMoviesData';
-import MainContextProvider from '../MainContextProvider/MainContextProvider';
+import MainContext from '../MainContext/MainContext';
 import ModalsWrapper from '../ModalsWrapper/ModalsWrapper';
 import { modalType } from '../Modal/Modal';
+import { BrowserRouter, Switch, Route } from 'react-router-dom';
 
 const MainContentWithWrapper = WithWrapper(MainContent);
-class App extends React.Component {
-  state = {
-    moviesList: [],
-    isLoading: true,
-    currentModal: modalType.none,
-    isModalShown: false
-  }
-
-  #openModal(modalType) {
-    this.setState({ isModalShown: true, currentModal: modalType });
-  }
-
-  #closeModal() {
-    this.setState({ isModalShown: false, currentModal: modalType.none });
-  }
-
-  render() {
-    const {
-      isLoading,
-      moviesList,
-      currentModal,
-      isModalShown
-    } = this.state;
-
-    return (
-      <MainContextProvider openModal={this.#openModal.bind(this)} closeModal={this.#closeModal.bind(this)} isLoading={isLoading} moviesList={moviesList}>
-        <div className={styles.App}>
-          <Header />
-          <MainContentWithWrapper />
-          <Footer />
-          <ModalsWrapper currentModal={currentModal} isModalShown={isModalShown} />
-        </div>
-      </MainContextProvider>
-    );
-  }
 
-  async componentDidMount() {
+const App = () => {
+  const [moviesList, setMoviesList] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [currentModal, setCurrentModal] = useState(modalType.none);
+  const [isModalShown, setIsModalShown] = useState(false);
+
+  const escapeHandler = e => e.which === 27 && isModalShown ? closeModal() : void 0;
+
+  const openModal = useCallback(modalType => {
+    setCurrentModal(modalType);
+    setIsModalShown(true);
+  }, [currentModal, isModalShown]);
+
+  const closeModal = useCallback(() => {
+    setCurrentModal(modalType.none);
+    setIsModalShown(false);
+  }, [currentModal, isModalShown]);
```

Comment:
after reading this article (https://kentcdodds.com/blog/usememo-and-usecallback) do you have some thoughts about relevanse of `useCallback` here? Thoughts?